addOns/openapi/src/test/java/org/zaproxy/zap/extension/openapi/ExtensionOpenApiTest.java

```diff
@@ -0,0 +1,261 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.openapi;
+
+import static fi.iki.elonen.NanoHTTPD.newFixedLengthResponse;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.withSettings;
+
+import fi.iki.elonen.NanoHTTPD;
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.nio.charset.Charset;
+import java.util.List;
+import org.apache.commons.httpclient.URI;
+import org.apache.commons.httpclient.URIException;
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.platform.commons.support.HierarchyTraversalMode;
+import org.junit.platform.commons.support.ReflectionSupport;
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.extension.ExtensionLoader;
+import org.parosproxy.paros.model.Model;
+import org.zaproxy.zap.extension.spider.ExtensionSpider;
+import org.zaproxy.zap.model.Context;
+import org.zaproxy.zap.testutils.NanoServerHandler;
+
+public class ExtensionOpenApiTest extends AbstractServerTest {
+
+    private ExtensionOpenApi classUnderTest;
+
+    @BeforeEach
+    public void setup() throws Exception {
+        classUnderTest = new ExtensionOpenApi();
+
+        Control control = mock(Control.class, withSettings().lenient());
+        setControlSingleton(control);
+
+        ExtensionLoader extensionLoader = mock(ExtensionLoader.class, withSettings().lenient());
+        when(control.getExtensionLoader()).thenReturn(extensionLoader);
+
+        given(extensionLoader.getExtension(ExtensionSpider.class))
+                .willReturn(mock(ExtensionSpider.class));
+    }
+
+    @Test
+    public void shouldImportFile() throws IOException {
+        // given
+        String test = "/PetStoreJson/";
+        String defnName = "defn.json";
+        this.nano.addHandler(new DefnServerHandler(test, defnName, "PetStore_defn.json"));
+        String fileContents =
+                getHtml(
+                        "PetStore_defn.json",
+                        new String[][] {{"PORT", String.valueOf(this.nano.getListeningPort())}});
+        File tmpFile = File.createTempFile("foo", "bar");
+        FileUtils.write(tmpFile, fileContents, Charset.defaultCharset());
+
+        // when
+        List<String> errors = classUnderTest.importOpenApiDefinition(tmpFile, false);
+
+        // then
+        assertThat("Should parse OK: " + errors, errors.isEmpty());
```

Comment:
It's preferable to use matchers which already provide a (good) error message than manually add a reason everywhere.